sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserverIntegration.java

```diff
@@ -15,21 +15,20 @@
 /** Watches the envelope dir. and send them (events) over. */
 public abstract class EnvelopeFileObserverIntegration implements Integration, Closeable {
   private @Nullable EnvelopeFileObserver observer;
-  private final @NotNull IEnvelopeReader envelopeReader;
+  private @Nullable ILogger logger;
 
-  EnvelopeFileObserverIntegration(final @NotNull IEnvelopeReader envelopeReader) {
-    this.envelopeReader = envelopeReader;
-  }
-
-  public static EnvelopeFileObserverIntegration getOutboxFileObserver(
-      final @NotNull IEnvelopeReader envelopeReader) {
-    return new OutboxEnvelopeFileObserverIntegration(envelopeReader);
+  public static @NotNull EnvelopeFileObserverIntegration getOutboxFileObserver() {
+    return new OutboxEnvelopeFileObserverIntegration();
   }
 
   @Override
-  public final void register(IHub hub, SentryOptions options) {
-    ILogger logger = options.getLogger();
-    String path = getPath(options);
+  public final void register(final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    Objects.requireNonNull(hub, "Hub is required");
```

Comment:
Hard to review this since it's just refactoring unrelated to the change

Reply:
this class needed to be changed, so it made sense to "improve" the missing finals and annotations IMO.
this class has more changes than this and adding NotNull helps with interoperability with Kotlin tests